src/v/kafka/server/group.cc

```diff
@@ -3273,7 +3273,7 @@ group::do_try_abort_old_tx(model::producer_identity pid) {
         }
         auto r = co_await _tx_frontend.local().try_abort(
           tm,
-          p_it->second.pid,
+          pid,
```

Comment:
It fixes the current case but the uninitialized field is still there and it may hurt us again. Possible options:
 - initialize it
 - remove it
 - if for some reason either option is available - document it, rename the field to `danger_do_not_use` etc.

In this case when `prepared_tx` comes from disk by replaying the log we set `.pid` (check group_recovery_consumer & group_stm) but when we replicate `prepared_tx` on the hot path and manually sync the in-memory state we miss to update `.pid` (see `store_txn_offsets`).

imho the best course of action is to fix `store_txn_offsets` to set `.pid`